src/utils/tcr.js

```diff
@@ -0,0 +1,99 @@
+const executePending = async ({
+  batchedSend,
+  fromBlock,
+  tcrContract,
+  toBN,
+  type
+}) => {
+  const statusChangeEvents = await tcrContract.getPastEvents(
+    `${type}StatusChange`,
+    { fromBlock }
+  )
+
+  // Take most recent event for each item.
+  const latestStatusChanges = {}
+  statusChangeEvents.forEach(event => {
+    const itemKey = event.raw.topics[3]
+    if (!latestStatusChanges[itemKey]) {
+      latestStatusChanges[itemKey] = event
+      return
+    }
+    if (event.blockNumber > latestStatusChanges[itemKey].blockNumber)
+      latestStatusChanges[itemKey] = event
+  })
+
+  const undisputedRequests = Object.keys(latestStatusChanges)
+    .map(itemKey => latestStatusChanges[itemKey])
+    .filter(requestEvent => {
+      // Only take status change events emitted when starting a request.
+      const {
+        returnValues: { _status }
+      } = requestEvent
+      return Number(_status) > 1 // Statuses 2 and 3 are registration and removal requested.
+    })
+    .filter(requestEvent => {
+      // Only take requests that were never challenged.
+      const {
+        raw: { topics }
+      } = requestEvent
+      const challenger = topics[2]
```

Comment:
You can use `returnValues`

Reply:
Same as in https://github.com/kleros/action-callback-bots/pull/1#discussion_r282968344. `returnValues` are incorrect for some reason.